src/main.rs

```diff
@@ -157,3 +244,47 @@ fn parse_line(line: &str) -> anyhow::Result<Option<DVec3>> {
         _ => Ok(None),
     }
 }
+
+fn histogram(
+    positions: &Vec<DVec3>,
+    r_variable: fn(&DVec3) -> f64,
+    (avg, std_dev): (DVec3, DVec3),
+) -> (Vec<Vec<DVec3>>, Vec<(f64, f64)>) {
+    let mut position_set: Vec<DVec3> = positions.clone();
```

Comment:
If you're going to clone a parameter right after receiving it, let the outside caller do the clone instead so that it's more visible. Take `positions` by value instead of by reference.